[release/public-v2] added a module build_linux_gnu, updated build_macos_gnu

natalie-perlin opened the pull request:
## ISSUE : 
Added a modulefile for Linux build_linux_gnu, 
Updated a modulelefile build_macos_gnu

## CONTRIBUTORS (optional): 
@gspetro-NOAA

natalie-perlin committed:
added a module build_linux_gnu, updated build_macos_gnu

dshawul commented on modulefiles/build_linux_gnu:
```diff
@@ -0,0 +1,87 @@
+module-whatis "Loads libraries needed for building SRW on Linux"
+
+if { [module-info mode load] } {
+   puts "ulimit -S -s unlimited;"
```
I found in the past, on a linux machine like hera, `system` has the desired effect instead of `puts`.
Could you check if you get unlimited stack after module loading this file on your system.
Same should go for puts statement in `wflow_linux`

mark-a-potts replied:
I can confirm that the puts works as expected (ulimit -s is unlimited after module load) on my Linux box. Since these are supposed to be templates, I think it might be a good idea to comment out almost everything and put in a stderr message that tells users to modify the file(s) to suit their system.

danielabdi-noaa replied:
I think it is the Lmod version that is the problem. On an ubuntu 18.04 with Lmod 6.6, puts simply prints the command to the screen whether you puts to stdout/stderr, while system has the required effect. It was the opposite for macos though. If the install instructions for Lmod on linux systems lead to the desired behaviour, I guess there will be no problem so I will approve.

natalie-perlin committed:
updated modulefiles for linux and macos

natalie-perlin committed:
added rocoto module loading template to linux and macos wflow_x modules

natalie-perlin committed:
updated wflow_linux and wflow_macos

Use env. variable $HOME to correspond the way the virtual environment was set during the build stage

natalie-perlin commented:
Corrected build_macos_gnu, build_linux_compiler (a template); 
Corrected wflow_macos, added wflow_linux
